Makefile

```
@@ -165,21 +174,38 @@ nix-purge: ##@nix Completely remove Nix setup, including /nix directory
 all: $(GO_CMD_NAMES)
 
 .PHONY: $(GO_CMD_NAMES) $(GO_CMD_PATHS) $(GO_CMD_BUILDS)
-$(GO_CMD_BUILDS): generate
+$(GO_CMD_BUILDS): generate $(LIBWAKU)
 $(GO_CMD_BUILDS): ##@build Build any Go project from cmd folder
+	CGO_LDFLAGS="$(CGO_LDFLAGS)" CGO_CFLAGS="$(CGO_CFLAGS)" \
 	go build -mod=vendor -v \
 		-tags '$(BUILD_TAGS)' $(BUILD_FLAGS) \
 		-o ./$@ ./cmd/$(notdir $@)
 	@echo "Compilation done."
 	@echo "Run \"build/bin/$(notdir $@) -h\" to view available commands."
 
-LIBWAKU := $(CURDIR)/vendor/github.com/waku-org/waku-go-bindings/third_party/nwaku/build/libwaku.$(LIBWAKU_EXT)
 $(LIBWAKU):
 ifeq ($(USE_NWAKU),true)
-	@echo "Building libwaku"
-	$(MAKE) -C $(CURDIR)/vendor/github.com/waku-org/waku-go-bindings/waku SHELL=/bin/bash
+	@echo "Building libwaku" $(LIBWAKU)
+	$(MAKE) -C $(NWAKU_SOURCE_DIR) SHELL=/bin/bash
 endif
 
+build-libwaku: $(LIBWAKU)
+
+test-libwaku: | $(LIBWAKU)
+	go test -tags '$(BUILD_TAGS) use_nwaku' -run TestDial ./wakuv2/... -count 1 -v -json | jq -r '.Output'
+
+clean-libwaku:
+	@echo "Removing libwaku"
+	rm $(LIBWAKU)
+
+rebuild-libwaku: | clean-libwaku $(LIBWAKU)
+
+clone-nwaku: NWAKU_SOURCE_DIR ?= $(GIT_ROOT)/../nwaku
+clone-nwaku: NWAKU_VERSION ?= v0.37.0-rc.3
+clone-nwaku: ##@build Clone or checkout nwaku v0.37.0-rc.3 into ../nwaku
+	@echo "Cloning nwaku $(NWAKU_VERSION)..."
+	git clone --branch $(NWAKU_VERSION) https://github.com/waku-org/nwaku.git $(NWAKU_SOURCE_DIR)
```

Why are you not utilizing the ability of `make` to skip targets that already exist?

```suggestion
NWAKU_SOURCE_DIR ?= $(GIT_ROOT)/../nwaku
NWAKU_VERSION ?= v0.37.0-rc.3

$(NWAKU_SOURCE_DIR): ##@build Clone or checkout nwaku v0.37.0-rc.3 into ../nwaku
	@echo "Cloning nwaku $(NWAKU_VERSION)..."
	git clone --branch $(NWAKU_VERSION) https://github.com/waku-org/nwaku.git $(NWAKU_SOURCE_DIR)

clone-nwaku: $(NWAKU_SOURCE_DIR)
```